src/app/gagestats/gagepage/gagepage.component.html

```diff
@@ -361,6 +325,88 @@
                                         <span *ngIf="newStat.isEditing"><textarea [(ngModel)]="newStat.comments" size="2"></textarea></span>
                                     </td>
 								</tr>
+								<ng-container *ngFor="let s of gage.statistics,  let sIndex = index">
+									<ng-container *ngIf="selectedStatGroup == 0 && (!preferred || s.isPreferred) || selectedStatGroup.includes(s.statisticGroupTypeID) && (!preferred || s.isPreferred)">
+										<tr *ngIf="statIds.includes(s.id)" >
```

Comment:
I'm curious why this is needed, looks like an unnecessary variable here, but I'm probably missing something!

Reply:
This might be overly complicated, but I is necessary. Line 328 is the for loop that reads through all the stats in a gage and creates a new row in the table for each one. Line 329 filters the stats by statGroupType and isPreferred. Line 330 adds the subheader with the statGroupType name (if its stat is included in the stat table).

Reply:
Oh, github displayed my comment weird - I was trying to just refer to line 330, talking about the `statIds` variable here. I was curious if it was needed, or if there's an already existing variable it could check, since this is the only place it's used

Reply:
The statIds is a list of the statistic ids where unique statisticGroupType occur for the first time. Kinda confusing (I could explain better in person). It is necessary to put the subheaders in the correct places.